book/writing-programs/cycle-tracking.md

```diff
@@ -56,69 +56,48 @@ fn main() {
 
-## Tracking Cycles with Tracing
+### Profiling a ZKVM program 
 
-The `cycle-tracker` annotation is a convenient way to track cycles for specific sections of code. However, sometimes it can also be useful to track what functions are taking the most cycles across the entire program, without having to annotate every function individually.
+Profiling a zkVM program produces a profile ([example link](https://share.firefox.dev/3Om1pzz)) which makes it easy to examine program performance and see exactly where exactly VM cycles are being spent without needing to modify the program at all.
 
-First, we need to generate a trace file of the program counter at each cycle while the program is executing. This can be done by simply setting the `TRACE_FILE` environment variable with the path of the file you want to write the trace to. For example, you can run the following command in the `script` directory for any example program:
+To profile a program, you simply need to:
+1. Enable the profiling feature for `sp1-sdk` in `script/Cargo.toml`
+2. Set the env variable `TRACE_FILE=trace.json` and then call `ProverClient::execute()` in your script.
 
-```bash
-TRACE_FILE=trace.log RUST_LOG=info cargo run --release
-```
+If you're executing a larger program (>100M cycles), you should set `TRACE_SAMPLE_RATE` to reduce the size of the trace file. A sample rate of `1000` means that 1 in every 1000 VM cycles is sampled. By default, every cycle is sampled.
 
-When the `TRACE_FILE` environment variable is set, as SP1's RISC-V runtime is executing, it will write a log of the program counter to the file specified by `TRACE_FILE`.
+Many examples can be found in the repo, such as this ['fibonacci'](https://github.com/succinctlabs/sp1/blob/12f212e386ae4c2da30cf6a61a7d87615d56bdac/examples/fibonacci/script/src/main.rs#L22) script.
 
-Next, we can use the `cargo prove` CLI with the `trace` command to analyze the trace file and generate a table of instruction counts. This can be done with the following command:
+Once you have your script it should look like the following: 
+```rs 
+    // Execute the program using the `ProverClient.execute` method, without generating a proof.
+    let (_, report) = client.execute(ELF, stdin.clone()).run().unwrap();
+```
 
-```bash
-cargo prove trace --elf <path_to_program_elf> --trace <path_to_trace_file>
+As well you must enable the profiling feature on the SDK:
+```toml
+    sp1-sdk = { version = "3.0.0", features = ["profiling"] }
 ```
 
-The `trace` command will generate a table of instruction counts, sorted by the number of cycles spent in each function. The output will look something like this:
+The `TRACE_FILE` env var tells the executor where to save the profile, and the `TRACE_SAMPLE_RATE` env var tells the executor how often to sample the program.
+A larger sample rate will give you a smaller profile, it is the number of instructions in between each sample.
 
+The full command to profile should look something like this
```

Comment:
Maybe we should just provide an end-to-end example of how to profile the fibonacci example or something. Right now these docs are a bit redundant. For example, I don't really like the overview section at the top. If you explain stuff step by step in your main explanation, I think that will probably suffice